### PR TITLE
[DF] Add test for friend alignment

### PR DESCRIPTION
**Pull request by vepadulano**
Sibling PR of https://github.com/root-project/root/pull/10939

### CODE REVIEW

File: python/distrdf/dask/check_friend_trees_alignment.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+TREENAMES = [f"distrdf_check_friend_trees_alignment_dask_tree_{i}.root" for i in range(1, 7)]
+FILENAMES = [f"distrdf_check_friend_trees_alignment_dask_file_{i}.root" for i in range(1, 7)]
+
+
+def create_dataset():
+    df = ROOT.RDataFrame(60).Define("x", "rdfentry_")
+
+    ranges = list(range(0, 61, 10))
+    for i in range(len(ranges[:-1])):
```
Comment (eguiraud) [inline python/distrdf/dask/check_friend_trees_alignment.py:17]: I think the python idiom for something like this is:

```python
for begin, end in zip(ranges, ranges[1:]):
```

File: python/distrdf/dask/check_friend_trees_alignment.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    yield
+    for name in FILENAMES:
+        os.remove(name)
+
+
+@pytest.mark.usefixtures("setup_testfriendtreesalignment")
+class TestDaskFriendTreesAlignment:
+    """Check that distrdf processes the correct entries from friend trees"""
+
+    def test_sum_friend_column(self, connection):
```
Comment (eguiraud) [inline python/distrdf/dask/check_friend_trees_alignment.py:55]: for my education, where does `connection` come from?

**Review by eguiraud — APPROVED** [review]
Thank you Vincenzo!

It's sad that we have to duplicate the whole test for spark and dask :/

File: python/distrdf/dask/check_friend_trees_alignment.py (hunk 2 above)
Comment (vepadulano) [reply python/distrdf/dask/check_friend_trees_alignment.py:55]: `connection` is a pytest fixture, creates a single Dask/Spark connection object for all the tests in the Dask/Spark suite

Comment (vepadulano) [reply python/distrdf/dask/check_friend_trees_alignment.py:55]: They are defined in the respective folders, for example the Dask one is at https://github.com/root-project/roottest/blob/master/python/distrdf/dask/conftest.py

**Pull request merged**